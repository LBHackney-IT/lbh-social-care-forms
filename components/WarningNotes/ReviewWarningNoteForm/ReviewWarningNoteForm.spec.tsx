import {
  fireEvent,
  render,
  RenderResult,
  screen,
  waitFor,
} from '@testing-library/react';
import React from 'react';
import { addYears, subMonths, subDays, addMonths, addDays } from 'date-fns';
import Router from 'next/router';

import { residentFactory } from 'factories/residents';
import { ReviewWarningNoteForm } from './ReviewWarningNoteForm';
import PersonView from '../../PersonView/PersonView';
import { AuthProvider } from '../../UserContext/UserContext';
import { mockedUser } from '../../../factories/users';
import {
  createMockedPersonView,
  setDateFieldValue,
} from '../../../test/helpers';
import * as warningNotes from '../../../utils/api/warningNotes';
import { mockedWarningNote } from '../../../factories/warningNotes';

jest.mock('../../PersonView/PersonView');
jest.mock('next/router', () => ({
  __esModule: true,
  default: {
    events: {
      on: jest.fn(),
    },
    push: jest.fn(),
  },
  useRouter: () => ({
    query: { foo: 'bar' },
    asPath: 'path',
    replace: jest.fn(),
    pathname: 'foopath',
  }),
}));
jest.mock('../../../utils/api/warningNotes');

const mockedAdultsResident = residentFactory.build({
  contextFlag: 'A',
});

const mockedChildrensResident = residentFactory.build({
  contextFlag: 'C',
});

describe('<ReviewWarningNoteForm />', () => {
  beforeEach(() => {
    jest.resetAllMocks();
  });

  describe('for any resident (adults or childrens)', () => {
    beforeEach(() => {
      (PersonView as jest.Mock).mockImplementation(
        createMockedPersonView(mockedAdultsResident)
      );

      jest.spyOn(warningNotes, 'useWarningNote').mockImplementation(() => ({
        data: mockedWarningNote[0],
        mutate: jest.fn(),
        revalidate: jest.fn(),
        isValidating: false,
      }));

      render(
        <AuthProvider user={mockedUser}>
          <ReviewWarningNoteForm personId={100} warningNoteId={10} />
        </AuthProvider>
      );
    });

    it('should show an error message if a review date is not entered', async () => {
      fireEvent.submit(screen.getByRole('form'));

      await waitFor(() => {
        screen.getByText('Enter a date review undertaken', {
          selector: '.govuk-error-message',
        });
      });
    });

    it('should show an error message if the review date entered is in the future', async () => {
      setDateFieldValue('reviewDate', addYears(new Date(), 2));

      fireEvent.submit(screen.getByRole('form'));

      await waitFor(() => {
        screen.getByText("Date review undertaken can't be in the future", {
          selector: '.govuk-error-message',
        });
      });
    });

    it('should show an error message if an invalid review date is entered', async () => {
      setDateFieldValue('reviewDate', {
        date: '56',
        month: '15',
        year: '10',
      });

      fireEvent.submit(screen.getByRole('form'));

      await waitFor(() => {
        screen.getByText('Must be a valid Date', {
          selector: '.govuk-error-message',
        });
      });
    });

    it('should show an error message if review details are not entered', async () => {
      fireEvent.submit(screen.getByRole('form'));

      await waitFor(() => {
        screen.getByText('Enter details of review', {
          selector: '.govuk-error-message',
        });
      });
    });

    it('should show an error message if the manager discussion date entered is in the future', async () => {
      setDateFieldValue('discussedWithManagerDate', addYears(new Date(), 2));

      fireEvent.submit(screen.getByRole('form'));

      await waitFor(() => {
        screen.getByText("Date discussed with manager can't be in the future", {
          selector: '.govuk-error-message',
        });
      });
    });

    it('should show an error message if the manager discussion date entered is invalid', async () => {
      setDateFieldValue('discussedWithManagerDate', {
        date: '56',
        month: '15',
        year: '10',
      });

      fireEvent.submit(screen.getByRole('form'));

      await waitFor(() => {
        screen.getByText('Must be a valid Date', {
          selector: '.govuk-error-message',
        });
      });
    });

    it('should show an error message if the manager discussion date entered is in the future', async () => {
      setDateFieldValue('discussedWithManagerDate', addYears(new Date(), 1));

      fireEvent.submit(screen.getByRole('form'));

      await waitFor(() => {
        screen.getByText("Date discussed with manager can't be in the future", {
          selector: '.govuk-error-message',
        });
      });
    });

    it('should show an error message if a next step is not selected', async () => {
      fireEvent.submit(screen.getByRole('form'));

      await waitFor(() => {
        screen.getByText('Select what you want to do', {
          selector: '.govuk-error-message',
        });
      });
    });

    it("should show an error message if the next step is selected as Renew and a next review date isn't entered", async () => {
      fireEvent.click(
        screen.getByText('Renew Warning Note', {
          selector: "[for='reviewDecision_Yes']",
        })
      );

      fireEvent.submit(screen.getByRole('form'));

      await waitFor(() => {
        screen.getByText('Enter a next review date', {
          selector: '.govuk-error-message',
        });
      });
    });

    it('should show an error message if the next step is selected as Renew and a next review date in the past is entered', async () => {
      fireEvent.click(
        screen.getByText('Renew Warning Note', {
          selector: "[for='reviewDecision_Yes']",
        })
      );

      setDateFieldValue('nextReviewDate', subMonths(new Date(), 2));

      fireEvent.submit(screen.getByRole('form'));

      await waitFor(() => {
        screen.getByText('Next review date cannot be earlier than today.', {
          selector: '.govuk-error-message',
        });
      });
    });

    it('should show an error message if the next step is selected as Renew and a next review date more than one year after the date review undertaken', async () => {
      fireEvent.click(
        screen.getByText('Renew Warning Note', {
          selector: "[for='reviewDecision_Yes']",
        })
      );

      setDateFieldValue('reviewDate', subDays(new Date(), 1)); // Yesterday
      setDateFieldValue('nextReviewDate', addDays(addYears(new Date(), 1), 1)); // One year and one day later

      fireEvent.submit(screen.getByRole('form'));

      await waitFor(() => {
        screen.getByText(
          'Next review date cannot be more than 1 year from date review undertaken.',
          {
            selector: '.govuk-error-message',
          }
        );
      });
    });

    describe('when the form is submitted with valid data', () => {
      it('should push to the summary route to renew a warning note', async () => {
        // Review date
        setDateFieldValue('reviewDate', new Date());

        // Discussed with individual
        fireEvent.click(
          screen.getByText('Yes', {
            selector: "[for='disclosedWithIndividual_Yes']",
          })
        );

        // Details of review
        fireEvent.change(screen.getByLabelText(/Details of review/), {
          target: {
            value: 'Some details',
          },
        });

        // Manager's name
        fireEvent.change(screen.getByLabelText(/Manager’s name/), {
          target: {
            value: 'First Last',
          },
        });

        // Date discussed with manager
        setDateFieldValue('discussedWithManagerDate', new Date());

        // Renew warning note
        fireEvent.click(
          screen.getByText('Renew Warning Note', {
            selector: "[for='reviewDecision_Yes']",
          })
        );

        // Next review date
        setDateFieldValue('nextReviewDate', addMonths(new Date(), 11));

        fireEvent.submit(screen.getByRole('form'));

        await waitFor(() => {
          expect(Router.push).toHaveBeenCalledWith(
            expect.objectContaining({
              pathname: '/people/100/warning-notes/10/summary',
            })
          );
        });
      });

      it('should push to the summary route to end a warning note', async () => {
        // Review date
        setDateFieldValue('reviewDate', new Date());

        // Discussed with individual
        fireEvent.click(
          screen.getByText('Yes', {
            selector: "[for='disclosedWithIndividual_Yes']",
          })
        );

        // Details of review
        fireEvent.change(screen.getByLabelText(/Details of review/), {
          target: {
            value: 'Some details',
          },
        });

        // Manager's name
        fireEvent.change(screen.getByLabelText(/Manager’s name/), {
          target: {
            value: 'First Last',
          },
        });

        // Date discussed with manager
        setDateFieldValue('discussedWithManagerDate', new Date());

        // End warning note
        fireEvent.click(
          screen.getByText('End Warning Note', {
            selector: "[for='reviewDecision_No']",
          })
        );

        fireEvent.submit(screen.getByRole('form'));

        await waitFor(() => {
          expect(Router.push).toHaveBeenCalledWith(
            expect.objectContaining({
              pathname: '/people/100/warning-notes/10/summary',
            })
          );
        });
      });
    });

    it('should not push to the summary route if there are errors on the form', async () => {
      fireEvent.submit(screen.getByRole('form'));

      await waitFor(() => {
        expect(Router.push).not.toHaveBeenCalled();
      });
    });
  });

  describe('for an adults resident', () => {
    let renderResult: RenderResult;

    beforeEach(() => {
      (PersonView as jest.Mock).mockImplementation(
        createMockedPersonView(mockedAdultsResident)
      );

      jest.spyOn(warningNotes, 'useWarningNote').mockImplementation(() => ({
        data: mockedWarningNote[0],
        mutate: jest.fn(),
        revalidate: jest.fn(),
        isValidating: false,
      }));

      renderResult = render(
        <AuthProvider user={mockedUser}>
          <ReviewWarningNoteForm personId={100} warningNoteId={10} />
        </AuthProvider>
      );
    });

    it('should render the warning notes form', () => {
      expect(renderResult.asFragment()).toMatchSnapshot();
    });

    it('should show an error message if a disclosure with individual value is not selected', async () => {
      fireEvent.submit(screen.getByRole('form'));

      await waitFor(() => {
        screen.getByText('Select an answer for this input', {
          selector: '.govuk-error-message',
        });
      });
    });

    it("should NOT show an error message if a manager's name is not entered", async () => {
      fireEvent.submit(screen.getByRole('form'));

      await waitFor(() => {
        expect(
          screen.queryByText('Enter a manager’s name', {
            selector: '.govuk-error-message',
          })
        ).toBeNull();
      });
    });

    it('should NOT show an error message if a manager discussion date is not entered', async () => {
      fireEvent.submit(screen.getByRole('form'));

      await waitFor(() => {
        expect(
          screen.queryByText('Enter a date discussed with manager', {
            selector: '.govuk-error-message',
          })
        ).toBeNull();
      });
    });
  });

  describe('for a childrens resident', () => {
    let renderResult: RenderResult;

    beforeEach(() => {
      (PersonView as jest.Mock).mockImplementation(
        createMockedPersonView(mockedChildrensResident)
      );

      jest.spyOn(warningNotes, 'useWarningNote').mockImplementation(() => ({
        data: mockedWarningNote[0],
        mutate: jest.fn(),
        revalidate: jest.fn(),
        isValidating: false,
      }));

      renderResult = render(
        <AuthProvider user={mockedUser}>
          <ReviewWarningNoteForm personId={100} warningNoteId={10} />
        </AuthProvider>
      );
    });

    it('should render the warning notes form', () => {
      expect(renderResult.asFragment()).toMatchSnapshot();
    });
  });
});
