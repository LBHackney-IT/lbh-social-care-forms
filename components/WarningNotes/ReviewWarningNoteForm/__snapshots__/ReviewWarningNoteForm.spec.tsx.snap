// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ReviewWarningNoteForm /> for a childrens resident should render the warning notes form 1`] = `
<DocumentFragment>
  <h1
    class="govuk-fieldset__legend--l gov-weight-lighter"
  >
    Review / end Warning Note
  </h1>
  <h2>
    Reviewing or ending a Warning Note
  </h2>
  <span
    class="govuk-caption-m"
  >
    Warnings must be kept under review by Team Managers and the evidence to support continued use must be reviewed at least annually. Warnings must always be reviewed on the closure or transfer of the case
  </span>
  <div
    class="govuk-!-margin-top-7"
  >
    <div
      class="govuk-!-margin-bottom-7"
    >
      <div
        class="lbh-table-header"
      >
        <div
          class="sectionTitle"
        >
          <h3
            class="govuk-heading-m govuk-custom-text-color govuk-!-margin-bottom-0"
          >
            WARNING DETAILS
          </h3>
        </div>
      </div>
      <hr
        class="govuk-divider"
      />
      <dl
        class="govuk-summary-list"
      >
        <div
          class="govuk-summary-list__row"
        >
          <dt
            class="govuk-summary-list__key"
          >
            Warning type
          </dt>
          <dd
            class="govuk-summary-list__value"
          >
            Risk to Adults
          </dd>
        </div>
        <div
          class="govuk-summary-list__row"
        >
          <dt
            class="govuk-summary-list__key"
          >
            Start date
          </dt>
          <dd
            class="govuk-summary-list__value"
          >
            22/11/2020
          </dd>
        </div>
        <div
          class="govuk-summary-list__row"
        >
          <dt
            class="govuk-summary-list__key"
          >
            Review / end date
          </dt>
          <dd
            class="govuk-summary-list__value"
          >
            01/12/2020
          </dd>
        </div>
        <div
          class="govuk-summary-list__row"
        >
          <dt
            class="govuk-summary-list__key"
          >
            Further details about the Warning Note
          </dt>
          <dd
            class="govuk-summary-list__value"
          >
            <div
              class="outer"
            >
              <p
                data-testid="para"
              >
                i am a note
              </p>
            </div>
          </dd>
        </div>
        <div
          class="govuk-summary-list__row"
        >
          <dt
            class="govuk-summary-list__key"
          >
            Manager’s name
          </dt>
          <dd
            class="govuk-summary-list__value"
          >
            the big boss
          </dd>
        </div>
        <div
          class="govuk-summary-list__row"
        >
          <dt
            class="govuk-summary-list__key"
          >
            Date discussed with manager
          </dt>
          <dd
            class="govuk-summary-list__value"
          >
            12/11/2020
          </dd>
        </div>
        <div
          class="govuk-summary-list__row"
        >
          <dt
            class="govuk-summary-list__key"
          >
            Created by
          </dt>
          <dd
            class="govuk-summary-list__value"
          >
            Foo
          </dd>
        </div>
      </dl>
    </div>
  </div>
  <h2>
    Warning Note review
  </h2>
  <div
    class="govuk-width-container"
  >
    <fieldset
      class="govuk-fieldset"
      role="group"
    >
      <form
        role="form"
      >
        <div
          class="govuk-form-group"
        >
          <div
            class="govuk-form-group"
          >
            <fieldset
              aria-describedby="reviewDate-hint"
              class="lbh-form-group govuk-form-group"
            >
              <legend
                class="govuk-label"
              >
                Date review undertaken 
                <span
                  class="govuk-required"
                >
                  *
                </span>
              </legend>
              <span
                class="lbh-hint"
                id="reviewDate-hint"
              />
              <div
                class="lbh-date-input govuk-date-input"
                id="reviewDate"
              >
                <div
                  class="lbh-date-input govuk-date-input__item"
                >
                  <div
                    class="lbh-form-group govuk-form-group"
                  >
                    <label
                      class="lbh-label govuk-date-input__label"
                      for="reviewDate-day"
                    >
                      Day
                    </label>
                    <input
                      class="lbh-date-input govuk-input govuk-input--width-2"
                      id="reviewDate-day"
                      inputmode="numeric"
                      name="reviewDate-day"
                      pattern="^\\\\d{2}$"
                      value=""
                    />
                  </div>
                </div>
                <div
                  class="lbh-date-input govuk-date-input__item"
                >
                  <div
                    class="lbh-form-group govuk-form-group"
                  >
                    <label
                      class="lbh-label govuk-date-input__label"
                      for="reviewDate-month"
                    >
                      Month
                    </label>
                    <input
                      class="lbh-input govuk-input govuk-date-input__input govuk-input--width-2"
                      id="reviewDate-month"
                      inputmode="numeric"
                      name="reviewDate-month"
                      pattern="^\\\\d{2}$"
                      value=""
                    />
                  </div>
                </div>
                <div
                  class="lbh-date-input govuk-date-input__item"
                >
                  <div
                    class="lbh-form-group govuk-form-group"
                  >
                    <label
                      class="lbh-label  govuk-date-input__label"
                      for="reviewDate-year"
                    >
                      Year
                    </label>
                    <input
                      class="lbh-input govuk-input govuk-date-input__input govuk-input--width-4"
                      id="reviewDate-year"
                      inputmode="numeric"
                      name="reviewDate-year"
                      pattern="^\\\\d{4}$"
                      value=""
                    />
                  </div>
                </div>
              </div>
            </fieldset>
          </div>
          <div
            class="govuk-form-group"
          >
            <div
              class="lbh-form-group govuk-form-group"
            >
              <label
                class="lbh-label govuk-label"
                for="reviewNotes"
              >
                Details of review 
                <span
                  class="govuk-required"
                >
                  *
                </span>
              </label>
              <span
                class="lbh-hint govuk-hint"
                id="reviewNotes-hint"
              >
                include details of disclosure to individual, any updates and why renewing or ending
              </span>
              <textarea
                aria-describedby="reviewNotes-hint"
                class="lbh-textarea govuk-textarea"
                id="reviewNotes"
                name="reviewNotes"
                rows="5"
              />
            </div>
          </div>
          <h3>
            Review discussed with manager
          </h3>
          <span
            class="govuk-caption-m"
          >
            This Warning Note review has been discussed and agreed by the manager named below
          </span>
          <div
            class="govuk-form-group"
          >
            <div
              class="govuk-form-group lbh-form-group"
            >
              <label
                class="govuk-label lbh-label"
                for="managerName"
              >
                Manager’s name 
                <span
                  class="govuk-required"
                >
                  *
                </span>
              </label>
              <input
                class="govuk-input lbh-input"
                data-testid="managerName"
                id="managerName"
                name="managerName"
                type="text"
              />
            </div>
          </div>
          <div
            class="govuk-form-group"
          >
            <fieldset
              aria-describedby="discussedWithManagerDate-hint"
              class="lbh-form-group govuk-form-group"
            >
              <legend
                class="govuk-label"
              >
                Date discussed with manager 
                <span
                  class="govuk-required"
                >
                  *
                </span>
              </legend>
              <span
                class="lbh-hint"
                id="discussedWithManagerDate-hint"
              />
              <div
                class="lbh-date-input govuk-date-input"
                id="discussedWithManagerDate"
              >
                <div
                  class="lbh-date-input govuk-date-input__item"
                >
                  <div
                    class="lbh-form-group govuk-form-group"
                  >
                    <label
                      class="lbh-label govuk-date-input__label"
                      for="discussedWithManagerDate-day"
                    >
                      Day
                    </label>
                    <input
                      class="lbh-date-input govuk-input govuk-input--width-2"
                      id="discussedWithManagerDate-day"
                      inputmode="numeric"
                      name="discussedWithManagerDate-day"
                      pattern="^\\\\d{2}$"
                      value=""
                    />
                  </div>
                </div>
                <div
                  class="lbh-date-input govuk-date-input__item"
                >
                  <div
                    class="lbh-form-group govuk-form-group"
                  >
                    <label
                      class="lbh-label govuk-date-input__label"
                      for="discussedWithManagerDate-month"
                    >
                      Month
                    </label>
                    <input
                      class="lbh-input govuk-input govuk-date-input__input govuk-input--width-2"
                      id="discussedWithManagerDate-month"
                      inputmode="numeric"
                      name="discussedWithManagerDate-month"
                      pattern="^\\\\d{2}$"
                      value=""
                    />
                  </div>
                </div>
                <div
                  class="lbh-date-input govuk-date-input__item"
                >
                  <div
                    class="lbh-form-group govuk-form-group"
                  >
                    <label
                      class="lbh-label  govuk-date-input__label"
                      for="discussedWithManagerDate-year"
                    >
                      Year
                    </label>
                    <input
                      class="lbh-input govuk-input govuk-date-input__input govuk-input--width-4"
                      id="discussedWithManagerDate-year"
                      inputmode="numeric"
                      name="discussedWithManagerDate-year"
                      pattern="^\\\\d{4}$"
                      value=""
                    />
                  </div>
                </div>
              </div>
            </fieldset>
          </div>
          <h3>
            Next steps
          </h3>
          <div
            class="govuk-form-group"
          >
            <div
              class="govuk-form-group lbh-form-group"
            >
              <label
                class="lbh-label govuk-label"
                for="reviewDecision"
              >
                What do you want to do? 
                <span
                  class="govuk-required"
                >
                  *
                </span>
              </label>
              <div
                class="govuk-radios lbh-radios"
              >
                <div
                  class="govuk-radios__item"
                >
                  <input
                    class="govuk-radios__input"
                    id="reviewDecision_Yes"
                    name="reviewDecision"
                    type="radio"
                    value="Yes"
                  />
                  <label
                    class="govuk-label govuk-radios__label"
                    for="reviewDecision_Yes"
                  >
                    Renew Warning Note
                  </label>
                </div>
                <div
                  class="govuk-radios__item"
                >
                  <input
                    class="govuk-radios__input"
                    id="reviewDecision_No"
                    name="reviewDecision"
                    type="radio"
                    value="No"
                  />
                  <label
                    class="govuk-label govuk-radios__label"
                    for="reviewDecision_No"
                  >
                    End Warning Note
                  </label>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="actions"
        >
          <button
            class="lbh-button govuk-button lbh-button--secondary lbh-button govuk-button"
            data-module="govuk-button"
            style="min-width: 200px;"
            type="button"
          >
            Save and finish later
          </button>
          <a
            class="lbh-link govuk-link"
            href="/"
            style="margin-left: 1.5rem;"
          >
            Cancel
          </a>
          <button
            class="lbh-button govuk-button"
            data-module="govuk-button"
            type="submit"
          >
            Continue
          </button>
        </div>
      </form>
    </fieldset>
  </div>
</DocumentFragment>
`;

exports[`<ReviewWarningNoteForm /> for an adults resident should render the warning notes form 1`] = `
<DocumentFragment>
  <h1
    class="govuk-fieldset__legend--l gov-weight-lighter"
  >
    Review / end Warning Note
  </h1>
  <h2>
    Reviewing or ending a Warning Note
  </h2>
  <span
    class="govuk-caption-m"
  >
    Warnings must be kept under review by Team Managers and the evidence to support continued use must be reviewed at least annually. Warnings must always be reviewed on the closure or transfer of the case
  </span>
  <div
    class="govuk-!-margin-top-7"
  >
    <div
      class="govuk-!-margin-bottom-7"
    >
      <div
        class="lbh-table-header"
      >
        <div
          class="sectionTitle"
        >
          <h3
            class="govuk-heading-m govuk-custom-text-color govuk-!-margin-bottom-0"
          >
            WARNING DETAILS
          </h3>
        </div>
      </div>
      <hr
        class="govuk-divider"
      />
      <dl
        class="govuk-summary-list"
      >
        <div
          class="govuk-summary-list__row"
        >
          <dt
            class="govuk-summary-list__key"
          >
            Warning type
          </dt>
          <dd
            class="govuk-summary-list__value"
          >
            Risk to Adults
          </dd>
        </div>
        <div
          class="govuk-summary-list__row"
        >
          <dt
            class="govuk-summary-list__key"
          >
            Start date
          </dt>
          <dd
            class="govuk-summary-list__value"
          >
            22/11/2020
          </dd>
        </div>
        <div
          class="govuk-summary-list__row"
        >
          <dt
            class="govuk-summary-list__key"
          >
            Review / end date
          </dt>
          <dd
            class="govuk-summary-list__value"
          >
            01/12/2020
          </dd>
        </div>
        <div
          class="govuk-summary-list__row"
        >
          <dt
            class="govuk-summary-list__key"
          >
            Is the individual aware of the Warning Note and Review Date?
          </dt>
          <dd
            class="govuk-summary-list__value"
          >
            Yes
          </dd>
        </div>
        <div
          class="govuk-summary-list__row"
        >
          <dt
            class="govuk-summary-list__key"
          >
            Date informed
          </dt>
          <dd
            class="govuk-summary-list__value"
          >
            12/11/2020
          </dd>
        </div>
        <div
          class="govuk-summary-list__row"
        >
          <dt
            class="govuk-summary-list__key"
          >
            How was the individual informed
          </dt>
          <dd
            class="govuk-summary-list__value"
          >
            Written
          </dd>
        </div>
        <div
          class="govuk-summary-list__row"
        >
          <dt
            class="govuk-summary-list__key"
          >
            Details of disclosure to individual
          </dt>
          <dd
            class="govuk-summary-list__value"
          >
            <div
              class="outer"
            >
              <p
                data-testid="para"
              >
                lorem ipsum
              </p>
            </div>
          </dd>
        </div>
        <div
          class="govuk-summary-list__row"
        >
          <dt
            class="govuk-summary-list__key"
          >
            Warning narrative and risks notes
          </dt>
          <dd
            class="govuk-summary-list__value"
          >
            <div
              class="outer"
            >
              <p
                data-testid="para"
              >
                i am a note
              </p>
            </div>
          </dd>
        </div>
        <div
          class="govuk-summary-list__row"
        >
          <dt
            class="govuk-summary-list__key"
          >
            Manager’s name
          </dt>
          <dd
            class="govuk-summary-list__value"
          >
            the big boss
          </dd>
        </div>
        <div
          class="govuk-summary-list__row"
        >
          <dt
            class="govuk-summary-list__key"
          >
            Date discussed with manager
          </dt>
          <dd
            class="govuk-summary-list__value"
          >
            12/11/2020
          </dd>
        </div>
        <div
          class="govuk-summary-list__row"
        >
          <dt
            class="govuk-summary-list__key"
          >
            Created by
          </dt>
          <dd
            class="govuk-summary-list__value"
          >
            Foo
          </dd>
        </div>
      </dl>
    </div>
  </div>
  <h2>
    Warning Note review
  </h2>
  <div
    class="govuk-width-container"
  >
    <fieldset
      class="govuk-fieldset"
      role="group"
    >
      <form
        role="form"
      >
        <div
          class="govuk-form-group"
        >
          <div
            class="govuk-form-group"
          >
            <fieldset
              aria-describedby="reviewDate-hint"
              class="lbh-form-group govuk-form-group"
            >
              <legend
                class="govuk-label"
              >
                Date review undertaken 
                <span
                  class="govuk-required"
                >
                  *
                </span>
              </legend>
              <span
                class="lbh-hint"
                id="reviewDate-hint"
              />
              <div
                class="lbh-date-input govuk-date-input"
                id="reviewDate"
              >
                <div
                  class="lbh-date-input govuk-date-input__item"
                >
                  <div
                    class="lbh-form-group govuk-form-group"
                  >
                    <label
                      class="lbh-label govuk-date-input__label"
                      for="reviewDate-day"
                    >
                      Day
                    </label>
                    <input
                      class="lbh-date-input govuk-input govuk-input--width-2"
                      id="reviewDate-day"
                      inputmode="numeric"
                      name="reviewDate-day"
                      pattern="^\\\\d{2}$"
                      value=""
                    />
                  </div>
                </div>
                <div
                  class="lbh-date-input govuk-date-input__item"
                >
                  <div
                    class="lbh-form-group govuk-form-group"
                  >
                    <label
                      class="lbh-label govuk-date-input__label"
                      for="reviewDate-month"
                    >
                      Month
                    </label>
                    <input
                      class="lbh-input govuk-input govuk-date-input__input govuk-input--width-2"
                      id="reviewDate-month"
                      inputmode="numeric"
                      name="reviewDate-month"
                      pattern="^\\\\d{2}$"
                      value=""
                    />
                  </div>
                </div>
                <div
                  class="lbh-date-input govuk-date-input__item"
                >
                  <div
                    class="lbh-form-group govuk-form-group"
                  >
                    <label
                      class="lbh-label  govuk-date-input__label"
                      for="reviewDate-year"
                    >
                      Year
                    </label>
                    <input
                      class="lbh-input govuk-input govuk-date-input__input govuk-input--width-4"
                      id="reviewDate-year"
                      inputmode="numeric"
                      name="reviewDate-year"
                      pattern="^\\\\d{4}$"
                      value=""
                    />
                  </div>
                </div>
              </div>
            </fieldset>
          </div>
          <div
            class="govuk-form-group"
          >
            <div
              class="govuk-form-group lbh-form-group"
            >
              <label
                class="lbh-label govuk-label"
                for="disclosedWithIndividual"
              >
                Have you discussed this review with the individual 
                <span
                  class="govuk-required"
                >
                  *
                </span>
              </label>
              <div
                class="govuk-radios lbh-radios"
              >
                <div
                  class="govuk-radios__item"
                >
                  <input
                    class="govuk-radios__input"
                    id="disclosedWithIndividual_Yes"
                    name="disclosedWithIndividual"
                    type="radio"
                    value="Yes"
                  />
                  <label
                    class="govuk-label govuk-radios__label"
                    for="disclosedWithIndividual_Yes"
                  >
                    Yes
                  </label>
                </div>
                <div
                  class="govuk-radios__item"
                >
                  <input
                    class="govuk-radios__input"
                    id="disclosedWithIndividual_No"
                    name="disclosedWithIndividual"
                    type="radio"
                    value="No"
                  />
                  <label
                    class="govuk-label govuk-radios__label"
                    for="disclosedWithIndividual_No"
                  >
                    No
                  </label>
                </div>
              </div>
            </div>
          </div>
          <div
            class="govuk-form-group"
          >
            <div
              class="lbh-form-group govuk-form-group"
            >
              <label
                class="lbh-label govuk-label"
                for="reviewNotes"
              >
                Details of review 
                <span
                  class="govuk-required"
                >
                  *
                </span>
              </label>
              <span
                class="lbh-hint govuk-hint"
                id="reviewNotes-hint"
              >
                include details of disclosure to individual, any updates and why renewing or ending
              </span>
              <textarea
                aria-describedby="reviewNotes-hint"
                class="lbh-textarea govuk-textarea"
                id="reviewNotes"
                name="reviewNotes"
                rows="5"
              />
            </div>
          </div>
          <h3>
            Review discussed with manager
          </h3>
          <span
            class="govuk-caption-m"
          >
            This Warning Note review has been discussed and agreed by the manager named below
          </span>
          <div
            class="govuk-form-group"
          >
            <div
              class="govuk-form-group lbh-form-group"
            >
              <label
                class="govuk-label lbh-label"
                for="managerName"
              >
                Manager’s name 
              </label>
              <input
                class="govuk-input lbh-input"
                data-testid="managerName"
                id="managerName"
                name="managerName"
                type="text"
              />
            </div>
          </div>
          <div
            class="govuk-form-group"
          >
            <fieldset
              aria-describedby="discussedWithManagerDate-hint"
              class="lbh-form-group govuk-form-group"
            >
              <legend
                class="govuk-label"
              >
                Date discussed with manager 
              </legend>
              <span
                class="lbh-hint"
                id="discussedWithManagerDate-hint"
              />
              <div
                class="lbh-date-input govuk-date-input"
                id="discussedWithManagerDate"
              >
                <div
                  class="lbh-date-input govuk-date-input__item"
                >
                  <div
                    class="lbh-form-group govuk-form-group"
                  >
                    <label
                      class="lbh-label govuk-date-input__label"
                      for="discussedWithManagerDate-day"
                    >
                      Day
                    </label>
                    <input
                      class="lbh-date-input govuk-input govuk-input--width-2"
                      id="discussedWithManagerDate-day"
                      inputmode="numeric"
                      name="discussedWithManagerDate-day"
                      pattern="^\\\\d{2}$"
                      value=""
                    />
                  </div>
                </div>
                <div
                  class="lbh-date-input govuk-date-input__item"
                >
                  <div
                    class="lbh-form-group govuk-form-group"
                  >
                    <label
                      class="lbh-label govuk-date-input__label"
                      for="discussedWithManagerDate-month"
                    >
                      Month
                    </label>
                    <input
                      class="lbh-input govuk-input govuk-date-input__input govuk-input--width-2"
                      id="discussedWithManagerDate-month"
                      inputmode="numeric"
                      name="discussedWithManagerDate-month"
                      pattern="^\\\\d{2}$"
                      value=""
                    />
                  </div>
                </div>
                <div
                  class="lbh-date-input govuk-date-input__item"
                >
                  <div
                    class="lbh-form-group govuk-form-group"
                  >
                    <label
                      class="lbh-label  govuk-date-input__label"
                      for="discussedWithManagerDate-year"
                    >
                      Year
                    </label>
                    <input
                      class="lbh-input govuk-input govuk-date-input__input govuk-input--width-4"
                      id="discussedWithManagerDate-year"
                      inputmode="numeric"
                      name="discussedWithManagerDate-year"
                      pattern="^\\\\d{4}$"
                      value=""
                    />
                  </div>
                </div>
              </div>
            </fieldset>
          </div>
          <h3>
            Next steps
          </h3>
          <div
            class="govuk-form-group"
          >
            <div
              class="govuk-form-group lbh-form-group"
            >
              <label
                class="lbh-label govuk-label"
                for="reviewDecision"
              >
                What do you want to do? 
                <span
                  class="govuk-required"
                >
                  *
                </span>
              </label>
              <div
                class="govuk-radios lbh-radios"
              >
                <div
                  class="govuk-radios__item"
                >
                  <input
                    class="govuk-radios__input"
                    id="reviewDecision_Yes"
                    name="reviewDecision"
                    type="radio"
                    value="Yes"
                  />
                  <label
                    class="govuk-label govuk-radios__label"
                    for="reviewDecision_Yes"
                  >
                    Renew Warning Note
                  </label>
                </div>
                <div
                  class="govuk-radios__item"
                >
                  <input
                    class="govuk-radios__input"
                    id="reviewDecision_No"
                    name="reviewDecision"
                    type="radio"
                    value="No"
                  />
                  <label
                    class="govuk-label govuk-radios__label"
                    for="reviewDecision_No"
                  >
                    End Warning Note
                  </label>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="actions"
        >
          <button
            class="lbh-button govuk-button lbh-button--secondary lbh-button govuk-button"
            data-module="govuk-button"
            style="min-width: 200px;"
            type="button"
          >
            Save and finish later
          </button>
          <a
            class="lbh-link govuk-link"
            href="/"
            style="margin-left: 1.5rem;"
          >
            Cancel
          </a>
          <button
            class="lbh-button govuk-button"
            data-module="govuk-button"
            type="submit"
          >
            Continue
          </button>
        </div>
      </form>
    </fieldset>
  </div>
</DocumentFragment>
`;
